ports/stm32/rfcore.c

```diff
@@ -589,18 +600,38 @@ static const struct {
 void rfcore_ble_init(void) {
     DEBUG_printf("rfcore_ble_init\n");
 
-    // Clear any outstanding messages from ipcc_init.
-    tl_check_msg(&ipcc_mem_sys_queue, IPCC_CH_SYS, NULL);
-
     // Configure and reset the BLE controller.
-    tl_sys_hci_cmd_resp(HCI_OPCODE(OGF_VENDOR, OCF_BLE_INIT), (const uint8_t *)&ble_init_params, sizeof(ble_init_params), 0);
-    tl_ble_hci_cmd_resp(HCI_OPCODE(0x03, 0x0003), NULL, 0);
+    if (!rfcore_ble_reset()) {
+        // ble init can fail if core2 has previously locked up. Reset HSI & rfcore to retry.
+        __HAL_RCC_HSI_DISABLE();
+        mp_hal_delay_ms(100);
+        __HAL_RCC_HSI_ENABLE();
+
+        rfcore_init();
```

Comment:
I found a reason to keep an extended retry in - another team at work has had intermittent issues with this same "timeout" symptom on new WB55 chips after they're initially programmed. Only happens on some units, not on most. I suspect there's a similar timing / race-condition of HSI clock on/off/on during programming then restart, first boot.

I've extended this retry to also include a HSI off, wait, on again. 

I got my board into the "permanently locked up state" where it would show timeout on `BLE().active(1)` even after hard reset. Then reflashed with this change and  `BLE().active(1)` now works, after showing a single instance of the timeout message.

Reply:
OK, let's keep it